sdk/tests/conformance2/textures/misc/tex-3d-size-limit.html

```diff
@@ -0,0 +1,117 @@
+** CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+** TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+-->
+
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="utf-8">
+<title>WebGL FramebufferTextureLayer Test</title>
+<link rel="stylesheet" href="../../../resources/js-test-style.css"/>
+<script src="../../../js/js-test-pre.js"></script>
+<script src="../../../js/webgl-test-utils.js"></script>
+</head>
+<body>
+<div id="description"></div>
+<div id="console"></div>
+<canvas id="canvas" width="2" height="2"> </canvas>
+
+<script>
+"use strict";
+var wtu = WebGLTestUtils;
+var gl;
+var canvas = document.getElementById("canvas");
+
+function numLevelsFromSize(size) {
+    var levels = 0;
+    while ((size >> levels) > 0) {
+        ++levels;
+    }
+    return levels;
+}
+
+function test3DTextureDimensions() {
+    debug("");
+    debug("Checking 3D texture dimensions.");
+
+    var tex3d = gl.createTexture();
+    gl.bindTexture(gl.TEXTURE_3D, tex3d);
+    var maxTexSize = gl.getParameter(gl.MAX_3D_TEXTURE_SIZE);
+    var maxLevels = numLevelsFromSize(maxTexSize);
+    var pixels = new Uint8Array(maxTexSize * 2 * 2 * 4);
+
+    gl.texImage3D(gl.TEXTURE_3D,
+                  0,                                // level
+                  gl.RGBA,                          // internalFormat
+                  1,                                // width
+                  1,                                // height
+                  -1,                               // depth
+                  0,                                // border
+                  gl.RGBA,                          // format
+                  gl.UNSIGNED_BYTE,                 // type
+                  pixels);                          // data
+    wtu.glErrorShouldBe(gl, gl.INVALID_VALUE,
+        "texImage3D should fail for dimension out of range.");
+    gl.texImage3D(gl.TEXTURE_3D, 0, gl.RGBA, maxTexSize + 1, 2, 2, 0, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
+    wtu.glErrorShouldBe(gl, gl.INVALID_VALUE,
+        "texImage3D should fail for dimension out of range.");
+
+    for (var i = 0; i < maxLevels; ++i) {
+        var size = 1 << i;
+        var level = maxLevels - i - 1;
+        var badSize = size * 2;
+
+        gl.texImage3D(gl.TEXTURE_3D, level, gl.RGBA, size, 1, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "texImage3D should succeed.");
+        gl.texImage3D(gl.TEXTURE_3D, level, gl.RGBA, 1, size, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "texImage3D should succeed.");
+        gl.texImage3D(gl.TEXTURE_3D, level, gl.RGBA, 1, 1, size, 0, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "texImage3D should succeed.");
+        gl.texImage3D(gl.TEXTURE_3D, level, gl.RGBA, badSize, 2, 2, 0, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
```

Comment:
Here your pixels data size is smaller than badSize * 2 * 2 * 4.  So it's kind of a mixed testing (pixels too small, and badSize).  I feel an isolated test case is much better.